release

```diff
@@ -0,0 +1,185 @@
+#!/bin/sh
+# vim: syntax=make ft=make
+#
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+MAKEFILE_START_LINE=$(\
+  grep -ne '^# 8><' "$0" \
+  | tail -n 1 \
+  | awk -F: '{print $1}')
+
+LITANI_RELEASE=tmp/litani-release
+mkdir -p "${LITANI_RELEASE}"
+
+MAKEFILE=${LITANI_RELEASE}/Makefile
+tail -n+${MAKEFILE_START_LINE} "$0" > "${MAKEFILE}"
+
+make -Bj 1 -f "${MAKEFILE}"
+SUCCESS="$?"
+
+exit "${SUCCESS}"
+
+# 8>< - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+#
+# Makefile starts here
+
+TMP=tmp/litani-release
+
+create-release: \
+	tac-installed \
+	checkout-develop \
+	$(TMP)/current-release-number \
+	$(TMP)/rc-number \
+	$(TMP)/commits.json \
+	merge-develop-release \
+	commit-changelog \
+	tag-release \
+	create-rc
+
+tac-installed:
+	$(info --- Testing for 'tac' installation)
+	printf '' | tac
+
+checkout-develop:
+	$(info --- Checking out develop)
+	git checkout develop
```

Comment:
When testing a release, I had to use: `git checkout --track origin/develop`

I got an error because my config was missing the following setting: `checkout.defaultRemote=origin`.

Sharing the full output:

```bash
$ ./release
--- Testing for 'tac' installation
printf '' | tac
--- Checking out develop
git checkout develop
hint: If you meant to check out a remote tracking branch on, e.g. 'origin',
hint: you can do so by fully qualifying the name with the --track option:
hint:
hint:     git checkout --track origin/<name>
hint:
hint: If you'd like to always have checkouts of an ambiguous <name> prefer
hint: one remote, e.g. the 'origin' remote, consider setting
hint: checkout.defaultRemote=origin in your config.
fatal: 'develop' matched multiple (3) remote tracking branches
make: *** [checkout-develop] Error 128
```

Reply:
I've added two new checks that run before the release process, the second one should hopefully address this comment:
1. Check that the remote you call `origin` actually is this GitHub awslabs repo
2. Check that the branch you call `develop` points to the same commit as `origin/develop`, and same for `release`.
One thing that it doesn't do is fetch from remote, I'm relying on the user to do that... maybe we should fetch just to be safe? But so far, this script works entirely locally and doesn't do anything with the internet at all, it relies on you to push also, and I think that's a nice property to have some how.

Reply:
My `origin` was indeed the awslabs repo and prior to running the script my `develop` pointed to the same commit as `origin/develop` (same for `release`). Prior to running the script, I also ran `git fetch --all` just to retrieve updates from multiple remotes (awslabs, yours, mine, Ronak's and Nick's).

I agree that not reaching out either to fetch or to push is best. In the latter case, we avoid executions of this script accidentally overwriting some branch.

Reply:
Are you absolutely sure that you had `develop` checked out when you ran this the first time? My understanding is that you wouldn't have got that error if you had an existing `develop` branch.

Would you mind running this script with a checked out `develop` pointing to `origin/develop` and try running this again? If it still doesn't work we shouldn't merge this

Reply:
> Are you absolutely sure that you had develop checked out when you ran this the first time?

You're right. I hadn't checked out develop. What I had done: 

1. Clone the repo.
2. Add remotes for your, mine, Nick's and Ronak's forks.
3. Fetch from all remotes.
4. Checkout `kk-release-script`
5. Run `./release`

What I did just now (step #4 is was what I originally hadn't done before):

1. `git clone git@github.com:awslabs/aws-build-accumulator.git`
2. `cd aws-build-accumulator`
3. `git fetch`
4.  `git checkout develop` # Got this message: `branch 'develop' set up to track 'origin/develop'. Switched to a new branch 'develop'`
6. Added all remotes and then `git fetch --all`
7. `git checkout kk-release-script`
8. `./release`

Reply:
Thanks. Then I think my addition this resolves this issue. The check
```
[ $$(git rev-parse develop) = $$(git rev-parse origin/develop) ]
```
will fail if step 4 had not been performed, or if for any other reason `develop` is pointing somewhere else.